tests/Loader/CucumberNDJsonAstLoaderTest.php

```diff
@@ -0,0 +1,175 @@
+                                    [
+                                        'location' => ['line' => 555],
+                                        'keyword' => 'exa',
+                                        'tableHeader' => [
+                                            'location' => ['line' => 666],
+                                            'cells' => [
+                                                ['value' => 'A'],
+                                                ['value' => 'B'],
+                                            ],
+                                        ],
+                                        'tableBody' => [
+                                            [
+                                                'location' => ['line' => 777],
+                                                'cells' => [
+                                                    ['value' => 'A1'],
+                                                    ['value' => 'B1'],
+                                                ],
+                                            ],
+                                            [
+                                                'location' => ['line' => 888],
+                                                'cells' => [
+                                                    ['value' => 'A2'],
+                                                    ['value' => 'B2'],
+                                                ],
+                                            ],
+                                        ],
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ]);
+        $features = $this->loader->load($file);
+
+        $this->assertEquals(
+            [
+                new FeatureNode(
+                    null,
+                    'Feature with a valid Scenario',
+                    [],
+                    new BackgroundNode('Empty Background', [], 'bac', 222),
+                    [
+                        new ScenarioNode('Empty Scenario', [], [], 'sce', 333),
+                        new OutlineNode(null, [], [], [
+                            new ExampleTableNode([
+                                666 => ['A', 'B'],
+                                777 => ['A1', 'B1'],
+                                888 => ['A2', 'B2'],
+                            ], 'exa'),
+                        ], 'out', 444),
+                    ],
+                    'fea',
+                    'en',
+                    $file,
+                    111,
+                ),
+            ],
+            $features,
+        );
+    }
+
+    public function testOutlineTableHeaderIsRequired(): void
+    {
+        $this->expectException(NodeException::class);
+        $this->expectExceptionMessage('Table header is required, but none was specified for the example on line 3.');
```

Comment:
I would always prefer to put the `expectException` at the last possible point before the code that is expected to throw. So in this case I'd generate the cucumber as a variable first, then put the expectations immediately before `$this->loader->load($resource)`

This reduces risk of false positives (which I appreciate are limited here) but also makes the failure output clearer if there's an unexpected exception during the setup of the test.

Reply:
Typically I'm doing that sort of thing as well, but somehow I never thought about it for exception.. good catch.